### PR TITLE
HADOOP-9016: added unit test org.apache.hadoop.tools.TestHadoopArchives....

### PR DESCRIPTION
**Pull request by ivan-veselovsky**
Added test ...testReadFileContent() checking the actual
file reading from the Har archive.

### CODE REVIEW

File: hadoop-tools/hadoop-archives/src/test/java/org/apache/hadoop/tools/TestHadoopArchives.java [hunk 1]
```diff
@@ -176,14 +217,14 @@ public void testPathWithSpaces() throws Exception {
           archivePath.toString()
       };
       System.setProperty(HadoopArchives.TEST_HADOOP_ARCHIVES_JAR_PATH, HADOOP_ARCHIVES_JAR);
-      final HadoopArchives har = new HadoopArchives(mapred.createJobConf());
+      final Configuration conf2 = miniMRClientCluster.getConfig();
+      final HadoopArchives har = new HadoopArchives(/*mapred.createJobConf()*/conf2);
```
Comment (dennisyv) [inline hadoop-tools/hadoop-archives/src/test/java/org/apache/hadoop/tools/TestHadoopArchives.java]: remove comment ?

File: hadoop-tools/hadoop-archives/src/test/java/org/apache/hadoop/tools/TestHadoopArchives.java [hunk 2]
```diff
@@ -223,4 +264,153 @@ public void testPathWithSpaces() throws Exception {
...
+        if (read < 0) {
+          // end of stream:
+          if (readIntoBuffer > 0) {
+            baos.write(buffer, 0, readIntoBuffer);
+          }
+          return baos.toByteArray();
+        } else {
+          readIntoBuffer += read;
+          if (readIntoBuffer == buffer.length) {
+            // buffer is full, need to clean the buffer.
```
Comment (dennisyv) [inline hadoop-tools/hadoop-archives/src/test/java/org/apache/hadoop/tools/TestHadoopArchives.java:385]: does someone really needs such detailed comments?

Comment (ivan-veselovsky) [inline hadoop-tools/hadoop-archives/src/test/java/org/apache/hadoop/tools/TestHadoopArchives.java:385]: yes, i think so.

File: hadoop-tools/hadoop-archives/src/test/java/org/apache/hadoop/tools/TestHadoopArchives.java [hunk 3]
```diff
@@ -223,4 +264,153 @@ public void testPathWithSpaces() throws Exception {
...
+        buffer = new byte[1024];
+        baos = new ByteArrayOutputStream(buffer.length * 2);
+      } else {
+        buffer = new byte[available + 1];
+        baos = new ByteArrayOutputStream(available);
+      }
+      int readIntoBuffer = 0;
+      int read; 
+      while (true) {
+        read = fsdis.read(buffer, readIntoBuffer, buffer.length - readIntoBuffer);
```
Comment (dennisyv) [inline hadoop-tools/hadoop-archives/src/test/java/org/apache/hadoop/tools/TestHadoopArchives.java:375]: not sure what do you want to test (probably read(..) method) but there is readFully method

File: hadoop-tools/hadoop-archives/src/test/java/org/apache/hadoop/tools/TestHadoopArchives.java (hunk 2 above)
Comment (dennisyv) [inline hadoop-tools/hadoop-archives/src/test/java/org/apache/hadoop/tools/TestHadoopArchives.java:385]: leave it if you want  - but I think it is unnecessary - it is so basic construction

File: hadoop-tools/hadoop-archives/src/test/java/org/apache/hadoop/tools/TestHadoopArchives.java [hunk 4]
```diff
@@ -223,4 +264,153 @@ public void testPathWithSpaces() throws Exception {
...
+  }
+  
+  private static byte[] prepareBin() {
+    byte[] bb = new byte[77777];
+    for (int i=0; i<bb.length; i++) {
+      // Generate unique values, as possible:
+      double d = Math.log(i + 2);
+      long bits = Double.doubleToLongBits(d);
+      byte b = (byte)bits;
+      bb[i] = b;
```
Comment (dennisyv) [inline hadoop-tools/hadoop-archives/src/test/java/org/apache/hadoop/tools/TestHadoopArchives.java]: may be just 
bb[i] = (byte)i;

?

File: hadoop-tools/hadoop-archives/src/test/java/org/apache/hadoop/tools/TestHadoopArchives.java (hunk 3 above)
Comment (ivan-veselovsky) [inline hadoop-tools/hadoop-archives/src/test/java/org/apache/hadoop/tools/TestHadoopArchives.java:375]: yes, here we test read() method,  #readFully() may be tested separately.

File: hadoop-tools/hadoop-archives/src/test/java/org/apache/hadoop/tools/TestHadoopArchives.java (hunk 1 above)
Comment (ivan-veselovsky) [inline hadoop-tools/hadoop-archives/src/test/java/org/apache/hadoop/tools/TestHadoopArchives.java]: thanks, fixed.

File: hadoop-tools/hadoop-archives/src/test/java/org/apache/hadoop/tools/TestHadoopArchives.java (hunk 4 above)
Comment (ivan-veselovsky) [inline hadoop-tools/hadoop-archives/src/test/java/org/apache/hadoop/tools/TestHadoopArchives.java]: thanks, fixed.

**Pull request merged**